.github/workflows/version_tests.yaml

```
@@ -0,0 +1,121 @@
+name: kuttl tests
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+    paths-ignore:
+      - 'docs/**'
+      - 'CHANGELOG/**'
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - 'docs/**'
+      - 'CHANGELOG/**'
+
+jobs:
+  build_image:
+    name: Build image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        if: github.event_name == 'pull_request'
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/checkout@v2
+        if: github.event_name != 'pull_request'
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-  uses: docker/setup-buildx-action@v1
+      - name: Build Docker image
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: Dockerfile
+          context: .
+          push: false
+          tags: k8ssandra/k8ssandra-operator:latest
+          platforms: linux/amd64
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          outputs: type=docker,dest=/tmp/k8ssandra-k8ssandra-operator.tar
+      - name: Upload k8ssandra-operator image
+        uses: actions/upload-artifact@v2
+        with:
+          name: k8ssandra-operator
+          path: /tmp/k8ssandra-k8ssandra-operator.tar
+  run_kuttl_test:
+    runs-on: ubuntu-latest
+    needs: [build_image]
+    name: Run kuttl tests
+    strategy:
+      matrix:
+        k8s_version: ["1.21"]
+        kuttl-test:
+        - test-cassandra-versions
+        serverVersion: 
+        - 3.11.10
+        - 3.11.11
+        - 3.11.12
+        - 4.0.0
+        - 4.0.1
+        - 4.0.2
+        - 4.0.3
```

Changing the order allows to see which Cassandra version is tested on the summary screen of GHA.
Also, we didn't publish a 4.0.2 image and went straight to 4.0.3.
My suggestion would also be to test against the latest of each minor release. Testing all patch releases will not be sustainable in the long run.

```suggestion
      matrix:
        k8s_version: ["1.21"]
        serverVersion: 
        - 3.11.12
        - 4.0.3
        kuttl-test:
        - test-cassandra-versions
```

---

All good points/suggestions. I've implemented all of these.